sdk/attestation/azure-security-attestation/azure/security/attestation/_models.py

```diff
@@ -69,242 +82,248 @@ class PolicyCertificatesModificationResult(object):
-        self._is_debuggable = kwargs.pop("is_debuggable", None) #type:Union[bool, None]
-        self._product_id = kwargs.pop("product_id", None) #type:Union[int, None]
-        self._mr_enclave = kwargs.pop("mr_enclave", None) #type:Union[str, None]
-        self._mr_signer = kwargs.pop("mr_signer", None) #type:Union[str, None]
-        self._svn = kwargs.pop("svn", None) #type:Union[int, None]
-        self._enclave_held_data = kwargs.pop("enclave_held_data", None) #type:Union[bytes, None]
-        self._sgx_collateral = kwargs.pop("sgx_collateral", None) #type:Union[dict, None]
+
+        """
+        self._issuer = kwargs.pop("issuer")  # type: Union[str, None]
+        self._unique_identifier = kwargs.pop(
+            "unique_identifier", None
+        )  # type: Union[str, None]
+        self._nonce = kwargs.pop("nonce", None)  # type: Union[str, None]
+        self._version = kwargs.pop("version")  # type: str
+        self._runtime_claims = kwargs.pop(
+            "runtime_claims", None
+        )  # type: Union[Dict, None]
+        self._inittime_claims = kwargs.pop(
+            "inittime_claims", None
+        )  # type: Union[Dict, None]
+        self._policy_claims = kwargs.pop(
+            "policy_claims", None
+        )  # type: Union[Dict, None]
+        self._verifier_type = kwargs.pop("verifier_type")  # type: str
+        self._policy_signer = kwargs.pop(
+            "policy_signer", None
+        )  # type: Union[AttestationSigner, None]
+        self._policy_hash = kwargs.pop("policy_hash")  # type: str
+        self._is_debuggable = kwargs.pop("is_debuggable")  # type: bool
+        self._product_id = kwargs.pop("product_id")  # type: int
+        self._mr_enclave = kwargs.pop("mr_enclave")  # type: str
+        self._mr_signer = kwargs.pop("mr_signer")  # type: str
+        self._svn = kwargs.pop("svn")  # type: int
+        self._enclave_held_data = kwargs.pop(
+            "enclave_held_data", None
+        )  # type: Union[bytes, None]
+        self._sgx_collateral = kwargs.pop("sgx_collateral")  # type: Dict
 
     @classmethod
     def _from_generated(cls, generated):
-        #type:(GeneratedAttestationResult) -> AttestationResult
+        # type: (GeneratedAttestationResult) -> AttestationResult
         return AttestationResult(
             issuer=generated.iss,
-            confirmation=generated.cnf,
             unique_identifier=generated.jti,
             nonce=generated.nonce,
             version=generated.version,
             runtime_claims=generated.runtime_claims,
             inittime_claims=generated.inittime_claims,
             policy_claims=generated.policy_claims,
             verifier_type=generated.verifier_type,
-            policy_signer=AttestationSigner._from_generated(generated.policy_signer) if generated.policy_signer else None,
+            policy_signer=AttestationSigner._from_generated(generated.policy_signer)
```

Comment:
your `_from_generated` method should handle the `None` case, I think the `if/else` portion is unnecessary

Reply:
That only works if `_from_generated(None)` returns `None`. But based on the examples in the `Tables` SDK, the `_from_generated` pattern says that `_from_generated(None)` should return an instance of the referenced type, not `None`.

`Monitor`, and `Storage` has the same pattern as `Tables`.

Several other SDKs don't have a special case for an empty `generated`.

Having said that, the `AppConfiguration` and the `Search` SDKs return None if `generated` is None, an instance of the class otherwise.

So the treatment of `generated` is inconsistent within the various Azure SDKs. The overall consensus is that a `._from_generated` method should return an instance of its class, but there are at least two SDKs that use a different pattern that returns `None` if the `generated` is `None`.

Is there a definitive version of the pattern?

Reply:
Ok, I just hit a problem with the Tables/Monitor/Storage implementation. Switching to the AppConfiguration/Search model (`if not generated: return None`).